main.c

```diff
@@ -524,6 +528,139 @@ init_vt(struct vkcube *vc)
    return 0;
 }
 
+static int
+create_drm_image(struct vkcube *vc, struct vkcube_buffer *b,
+    uint64_t drm_format, uint64_t drm_format_mod)
+{
+      int fd;
+
+      b->gbm_bo = gbm_bo_create_with_modifiers(vc->gbm_device, vc->width, vc->height,
+                                drm_format, &drm_format_mod, 1);
+
+      fd = gbm_bo_get_fd(b->gbm_bo);
+
+      VkExternalImageFormatProperties external_image_format_props = {
+          .sType = VK_STRUCTURE_TYPE_EXTERNAL_IMAGE_FORMAT_PROPERTIES,
+      };
+      VkImageFormatProperties2 image_format_props = {
+          .sType = VK_STRUCTURE_TYPE_IMAGE_FORMAT_PROPERTIES_2,
+          .pNext = &external_image_format_props,
+      };
+      VkResult res = vkGetPhysicalDeviceImageFormatProperties2(vc->physical_device,
+                                                               &(VkPhysicalDeviceImageFormatInfo2) {
+                                                                   .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_IMAGE_FORMAT_INFO_2,
+                                                                   .pNext = &(VkPhysicalDeviceExternalImageFormatInfo){
+                                                                       .sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_EXTERNAL_IMAGE_FORMAT_INFO,
+                                                                       .handleType = VK_EXTERNAL_MEMORY_HANDLE_TYPE_DMA_BUF_BIT_EXT,
+                                                                   },
+                                                                   .format = vc->image_format,
+                                                                   .type = VK_IMAGE_TYPE_2D,
+                                                                   .tiling = VK_IMAGE_TILING_LINEAR,
+                                                                   .usage = VK_IMAGE_USAGE_COLOR_ATTACHMENT_BIT,
+                                                               },
+                                                               &image_format_props);
+      fail_if(res != VK_SUCCESS, "Failed to get DeviceImageFormatProperties.\n");
+
+      VkExternalMemoryFeatureFlags feature_flags = external_image_format_props
+          .externalMemoryProperties
+          .externalMemoryFeatures;
+      fail_if(!(feature_flags & VK_EXTERNAL_MEMORY_FEATURE_IMPORTABLE_BIT), "dma-buf does not support Vulkan import");
+
+      VkImageDrmFormatModifierExplicitCreateInfoEXT mod_create_info = {
+          .sType = VK_STRUCTURE_TYPE_IMAGE_DRM_FORMAT_MODIFIER_EXPLICIT_CREATE_INFO_EXT,
+          .pNext = &(VkExternalMemoryImageCreateInfo){
```

Comment:
Is this project autoformatter-friendly? If so you might still have to run it :)

```suggestion
          .pNext = &(VkExternalMemoryImageCreateInfo) {
```